patches/server/0005-Paper-config-files.patch

```diff
@@ -1841,6 +1841,12 @@ index 0000000000000000000000000000000000000000..d47c57afafc01e25b965f1844938b251
 +        public boolean portalSearchVanillaDimensionScaling = true;
 +        public boolean disableTeleportationSuffocationCheck = false;
 +        public IntOr.Disabled netherCeilingVoidDamageHeight = IntOr.Disabled.DISABLED;
++
++        public Mushroom mushroom;
++        public class Mushroom extends ConfigurationPart {
++            public BooleanOrDefault disableLimit = BooleanOrDefault.USE_DEFAULT;
++            public IntOr.Default spreadLimit = IntOr.Default.USE_DEFAULT;
```

Comment:
Like I mentioned in your previous PR, I think it would be better if this was combined into 1 setting. Then it wouldn't need a new section. You can create an IntOrDefaultOrDisabled which can be `default`, `disabled` or an integer. 

You can model that type based on the existing types, just add another constant string value that is valid.

Reply:
I can't do that for the moment. I'll ask someone on the discord.